### ISSUE TITLE
Rule proposal: `prefer-await`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#1572]
Similar to [`unicorn/prefer-top-level-await`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-top-level-await.md) but it should trigger on non-top-level usage of `.then`, `.catch` and `.finally` instead. There exists [`github/no-then`](https://github.com/github/eslint-plugin-github/blob/main/docs/rules/no-then.md) which works on all levels but I find the fact that it's not configurable for which level it's working on lacking.

There should be an option `topLevelAwait` (default `false`) which defines how top-level is handled, primarily to support legacy versions of Node.js which do not support top-level-await.

## Fail

```js
function bar() {
  foo().then(baz).catch(baz);
}
```

```js
function bar() {
  foo().finally(baz);
}
```

With option `topLevelAwait` set to `true`:

```js
foo().then(baz).catch(baz);
```

## Pass

```js
async function bar() {
  try {
    await foo();
    baz();  
  } catch {
    baz();
  }
}
```

With option `topLevelAwait` set to `false`:

```js
foo().then(baz).catch(baz);
```
